Added Circular Linked List

mdzeeshan-2 opened the pull request:
Please accept my pull request.

ShubhKotnala commented:
Hi @mdzeeshan-2 , Please add a comment block at the top of the code to explain what this code will do, with some test cases, which your code will validate using the driver code. 
Also, do format the code, removing extra spaces, and following right indentation. 

Do have a look at the Contributors guideline, which would have been actually present while creating this PR. 
I'm adding it back, please make sure your follow all the check points. 

Cheers.

ShubhKotnala commented on python/Circluar Linked List/LinkedList.py:
```diff
@@ -0,0 +1,93 @@
+# Python program to Search an Element in a Circular Linked List
+# Md Zeeshan
```
Explain more about the problem, with some test cases and validate those test cases with your drive code.

ShubhKotnala commented on CPP/KMP/KMP.cpp:
```diff
@@ -0,0 +1,74 @@
+// Program to implement the KMP algorithm in C++
+//Md Zeeshan
+
```
Explain more about the problem, with some test cases and validate those test cases with your drive code.

mdzeeshan-2 replied:
Bhaiya I added Version 2 of Circular Linked List. Please check it and command me if any changes is needed.

ShubhKotnala replied:
Explanation needs to be present in the code file. 

Let's keep it simple. 
Think of a person who is just starting with coding, and comes across this repository and finds many programs across multiple languages. Now how that person is supposed to know what is the actual functionality of the code he is currently viewing if it find only a code block. 
Seems logical?

Have a look at [this](https://github.com/ShubhKotnala/Data_Structures_And_Algorithms/blob/main/java/array/MountainArray.java) file, and see how it explains its reader what the program is going to before actually starting with the code.

mdzeeshan-2 commented:
<!--
/* Font Definitions */
@font-face
	{font-family:"Cambria Math";
	panose-1:2 4 5 3 5 4 6 3 2 4;}
@font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4;}
/* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{margin:0cm;
	font-size:11.0pt;
	font-family:"Calibri",sans-serif;}
a:link, span.MsoHyperlink
	{mso-style-priority:99;
	color:blue;
	text-decoration:underline;}
pre
	{mso-style-priority:99;
	mso-style-link:"HTML Preformatted Char";
	margin:0cm;
	font-size:10.0pt;
	font-family:"Courier New";}
span.HTMLPreformattedChar
	{mso-style-name:"HTML Preformatted Char";
	mso-style-priority:99;
	mso-style-link:"HTML Preformatted";
	font-family:"Courier New";}
.MsoChpDefault
	{mso-style-type:export-only;}
@page WordSection1
	{size:612.0pt 792.0pt;
	margin:72.0pt 72.0pt 72.0pt 72.0pt;}
div.WordSection1
	{page:WordSection1;}
-->Bhaiya Please recheck my pull request. I added Circular Linked List version 2 folder. Please that only, and please accept my pull request before the Hacktoberfest. Sent from Mail for Windows From: Shubhankar KotnalaSent: 28 October 2022 19:28To: ShubhKotnala/Data_Structures_And_AlgorithmsCc: Md Zeeshan; MentionSubject: Re: [ShubhKotnala/Data_Structures_And_Algorithms] Added Circular Linked List and KMP code (PR ***@***.*** requested changes on this pull request.In python/Circluar Linked List/LinkedList.py:> +# Python program to Search an Element in a Circular Linked List+# Md ZeeshanExplain more about the problem, with some test cases and validate those test cases with your drive code.In CPP/KMP/KMP.cpp:> +// Program to implement the KMP algorithm in C+++//Md Zeeshan+Explain more about the problem, with some test cases and validate those test cases with your drive code.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***> 

mdzeeshan-2 commented:
Bhaiya I just added "Circular Linked List Version 3" Please checkout the and instruct any changes if needed. 
Here i added full user understanding language before the code where anyone can know about Circular linked list before reaching out the codes like what happens in Circular Linked list and how does it works. Please check it out and Merge "Added Circular Linked List Version 3" folder with your repository.

mdzeeshan-2 commented:
<!--
/* Font Definitions */
@font-face
	{font-family:"Cambria Math";
	panose-1:2 4 5 3 5 4 6 3 2 4;}
@font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4;}
/* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{margin:0cm;
	font-size:11.0pt;
	font-family:"Calibri",sans-serif;}
a:link, span.MsoHyperlink
	{mso-style-priority:99;
	color:blue;
	text-decoration:underline;}
pre
	{mso-style-priority:99;
	mso-style-link:"HTML Preformatted Char";
	margin:0cm;
	font-size:10.0pt;
	font-family:"Courier New";}
span.HTMLPreformattedChar
	{mso-style-name:"HTML Preformatted Char";
	mso-style-priority:99;
	mso-style-link:"HTML Preformatted";
	font-family:"Courier New";}
.MsoChpDefault
	{mso-style-type:export-only;}
@page WordSection1
	{size:612.0pt 792.0pt;
	margin:72.0pt 72.0pt 72.0pt 72.0pt;}
div.WordSection1
	{page:WordSection1;}
-->Bhaiya I just added "Circular Linked List Version 3" Please checkout the and instruct any changes if needed.Here i added full user understanding language before the code where anyone can know about Circular linked list before reaching out the codes like what happens in Circular Linked list and how does it works. Please check it out and Merge "Added Circular Linked List Version 3" folder with your repository. Sent from Mail for Windows From: Shubhankar KotnalaSent: 29 October 2022 20:42To: ShubhKotnala/Data_Structures_And_AlgorithmsCc: Md Zeeshan; MentionSubject: Re: [ShubhKotnala/Data_Structures_And_Algorithms] Added Circular Linked List and KMP code (PR ***@***.*** commented on this pull request.In CPP/KMP/KMP.cpp:> +// Program to implement the KMP algorithm in C+++//Md Zeeshan+Explanation needs to be present in the code file.Let's keep it simple.Think of a person who is just starting with coding, and comes across this repository and finds many programs across multiple languages. Now how that person is supposed to know what is the actual functionality of the code he is currently viewing if it find only a code block.Seems logical?Have a look at this file, and see how it explains its reader what the program is going to before actually starting with the code.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***> 
